src/box/txn.c

```diff
@@ -301,7 +481,9 @@ txn_free(struct txn *txn)
 		txn_stmt_destroy(stmt);
 
 	/* Truncate region up to struct txn size. */
-	region_truncate(&txn->region, sizeof(struct txn));
+	txn_reset_stats(txn);
+	region_truncate(&txn->region,
+			sizeof(struct txn) + sizeof(uint32_t) * TX_ALLOC_MAX);
```

Comment:
If you use `uint32_t alloc_stats[TX_ALLOC_MAX];` you will not need to change this line.

Reply:
done